### PR TITLE
VMHostVss - VMHostVssShaping

### PR DESCRIPTION
**Pull request by lucdekens**
VSS traffic shaping resource

### CODE REVIEW

File: Source/VMware.vSphereDSC/Tests/Integration/VMHostVss/VMHostVssShaping.Integration.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+            }
+
+            It 'Should compile and apply the MOF without throwing' {
+                # Assert
+                $startDscConfigurationParameters = @{
+                    Path = $script:mofFileWithModifyVssShaping
+                    ComputerName = 'localhost'
+                    Wait = $true
+                    Force = $true
+                }
```
Comment (SimeonGerginov) [inline Source/VMware.vSphereDSC/Tests/Integration/VMHostVss/VMHostVssShaping.Integration.Tests.ps1]: 1. Add empty line here and also move the # Assert comment here.

```
 # Assert
 { Start-DscConfiguration @startDscConfigurationParameters } | Should -Not -Throw
```

The same comments for the other tests.

File: Documentation/DSCResources/VMHostVssShaping/VMHostVssShaping.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+$script:configurationData = @{
+    AllNodes = @(
+        @{
+            NodeName = 'localhost'
+            PSDscAllowPlainTextPassword = $true
+        }
+    )
+}
+
+Configuration VMHostVssShaping_Config {
```
[the remark this answers is not in the document]
Comment (lucdekens) [reply Documentation/DSCResources/VMHostVssShaping/VMHostVssShaping.md]: Changed

File: Source/VMware.vSphereDSC/Configurations/ESXiConfigs/VMHostVss/VMHostVssShaping_Config.ps1 [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+$script:configurationData = @{
+    AllNodes = @(
+        @{
+            NodeName = 'localhost'
+            PSDscAllowPlainTextPassword = $true
+        }
+    )
+}
+
+Configuration VMHostVssShaping_Config {
```
[the remark this answers is not in the document]
Comment (lucdekens) [reply Source/VMware.vSphereDSC/Configurations/ESXiConfigs/VMHostVss/VMHostVssShaping_Config.ps1]: Changed

File: Source/VMware.vSphereDSC/DSCResources/VMHostNetwork/VMHostVss/VMHostVssShaping.ps1 [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+            $vmHostVSSShaping.AverageBandwidth = $currentVss.Spec.Policy.ShapingPolicy.AverageBandwidth
+            $vmHostVSSShaping.BurstSize = $currentVss.Spec.Policy.ShapingPolicy.BurstSize
+            $vmHostVSSShaping.Enabled = $currentVss.Spec.Policy.ShapingPolicy.Enabled
+            $vmHostVSSShaping.PeakBandwidth = $currentVss.Spec.Policy.ShapingPolicy.PeakBandwidth
+        }
+        else {
+            $vmHostVSSShaping.VssName = $this.Name
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (lucdekens) [reply Source/VMware.vSphereDSC/DSCResources/VMHostNetwork/VMHostVss/VMHostVssShaping.ps1]: Changed

File: Source/VMware.vSphereDSC/DSCResources/VMHostNetwork/VMHostVss/VMHostVssShaping.ps1 [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+
+        $this.ConnectVIServer()
+        $vmHost = $this.GetVMHost()
+        $this.GetNetworkSystem($vmHost)
+        $vss = $this.GetVss()
+
+        if ($this.Ensure -eq [Ensure]::Present) {
+            return ($null -ne $vss -and $this.Equals($vss))
+        }
+        else {
```
[the remark this answers is not in the document]
Comment (lucdekens) [reply Source/VMware.vSphereDSC/DSCResources/VMHostNetwork/VMHostVss/VMHostVssShaping.ps1]: Changed

File: Source/VMware.vSphereDSC/Tests/Integration/VMHostVss/VMHostVssShaping.Integration.Tests.ps1 (hunk 1 above)
Comment (lucdekens) [reply Source/VMware.vSphereDSC/Tests/Integration/VMHostVss/VMHostVssShaping.Integration.Tests.ps1]: Changed

File: Source/VMware.vSphereDSC/DSCResources/VMHostNetwork/VMHostVss/VMHostVssShaping.ps1 (hunk 5 above)
[the remark this answers is not in the document]
Comment (SimeonGerginov) [reply Source/VMware.vSphereDSC/DSCResources/VMHostNetwork/VMHostVss/VMHostVssShaping.ps1]: In the if clause remove the empty line:

```
 if ($this.Ensure -eq [Ensure]::Present) {
     return ($null -ne $vss -and $this.Equals($vss))
 }
```

File: Source/VMware.vSphereDSC/Tests/Integration/VMHostVss/VMHostVssShaping.Integration.Tests.ps1 (hunk 1 above)
Comment (SimeonGerginov) [reply Source/VMware.vSphereDSC/Tests/Integration/VMHostVss/VMHostVssShaping.Integration.Tests.ps1]: Do the same thing for the second Configuration tests.

File: Source/VMware.vSphereDSC/DSCResources/VMHostNetwork/VMHostVss/VMHostVssShaping.ps1 (hunk 5 above)
[the remark this answers is not in the document]
Comment (lucdekens) [reply Source/VMware.vSphereDSC/DSCResources/VMHostNetwork/VMHostVss/VMHostVssShaping.ps1]: Changed

File: Source/VMware.vSphereDSC/Tests/Integration/VMHostVss/VMHostVssShaping.Integration.Tests.ps1 (hunk 1 above)
Comment (lucdekens) [reply Source/VMware.vSphereDSC/Tests/Integration/VMHostVss/VMHostVssShaping.Integration.Tests.ps1]: Changed

**Pull request merged**